### ISSUE TITLE
Incorrect example for interactive plotting in Matplotlib Usage FAQ

### ISSUE THREAD

**Issue by vab9** [matplotlib/matplotlib#8883]
### Bug report

**Bug summary**

The interactive example in the Matplotlib FAQ does not work as described. 

**Code for reproduction**

I followed these very simple instructions from the [FAQ](https://matplotlib.org/faq/usage_faq.html#interactive-example):

```python
# in shell execute one by one:
import matplotlib.pyplot as plt

plt.ion()
plt.plot([1.6, 2.7])

# plot is shown

plt.title("interactive test")
plt.xlabel("index")

# plot is updated with titel and label

ax = plt.gca()
ax.plot([3.1, 2.2])

# (!) second line is added to the axes
# no need to call:
# plt.draw()
```

When I execute:

```python
# Paste your code here
ax = plt.gca()
ax.plot([3.1, 2.2])
```

my plot immediately updates with the new line. The FAQ claims this should not happen until I call `draw()`.

The same thing happens when I use the 'TKAgg' backend and not my default 'MacOSX'. I also tried this with both Python and iPython.

**Actual outcome**

Plot is updated with new line.

**Expected outcome**

Plot should not be updated until `draw()` is called.

**Matplotlib version**
  * Operating System: Mac OSX 10.11.6
  * Matplotlib Version: 2.0.2
  * Python Version: 3.6.1
  * Anaconda Version: 4.4.0
  * Jupyter Version (if applicable): 4.3.0

Installed matplotlib by installing Anaconda 4.4.0 (default channel).

**Comment by WeatherGod** [comment]
No, that is correct behavior for TkAgg. The point of interactive mode is to
not need to call plt.draw() or plt.show(). The macosx backend is faulty
here in this case.

By default, matplotlib does not have the interactive mode turned on, so you
would usually need to call draw() or show() explicitly if you want the
figure to appear or be updated. Perhaps the FAQ should be clarified?

**Comment by vab9** [comment]
Thanks for the quick reply! So is the FAQ wrong where it says:

> Axes methods do not include an automatic call to `draw_if_interactive()`; that call is added by the pyplot functions.

? Or am I misunderstanding that sentence somehow? Do you know if this behaviour is consistent across other interactive backends? 

Thanks for clarifying this for me. If you let me know where we can edit the FAQ, I'd be glad to open a PR...

**Comment by WeatherGod** [comment]
Ah, oops. It *used* to be that way before v1.5. Looks like we missed this
in the documentation. Now, any plotting call *should* update the display
(but it is a known problem that macosx doesn't do so for everything).

This can be updated in `tutorials/01_introductory/usage.py` on master.

**Comment by anntzer** [comment]
Closed by #8897, thanks for the fix.
